### DISCUSSION TITLE
How to setup the Activity Watch server on a single cloud server and let all the clients connect to it?

### DISCUSSION THREAD

**Discussion by IonicaBizau** [ActivityWatch/activitywatch#569] [Q&A]
Since I use several machines, I would prefer running the server in a centralized way.

How can I set up the URL of the server in the Chrome extension and in the client?

Thank you! ❇️

**Comment by pcuci** [comment]
Yes, this would be great.

I have multiple home machines, as well as multiple work machines (behind a work VPN). Some of these machines are GUI-less (so can we track remote ssh activity?)

Having an easy way to aggregate the whole of my activity online would be fun.

Not to mention mobile devices and my Fitbit... Was looking at https://www.humanapi.co/ for a way to integrate all these data streams, it doesn't look trivial.

PS: https://workrave.org/ seems to have figured out a way to track activity cross-platform across machines, seems LAN limited though.

**Comment by ErikBjare** [comment]
This has been discussed in numerous places before, and while we generally discourage doing it for security reasons, it is possible (and might even be reasonably secure if you do it by using SSH port forwarding instead of exposing aw-server on the network directly).

A search will lead to several posts on the forum where it has been discussed, here are a few:
 - https://forum.activitywatch.net/t/run-client-from-windows-ubuntu/685 (this one links to several others)
 - https://forum.activitywatch.net/t/using-activitywatch-in-an-enterprise-environment/290
 - https://forum.activitywatch.net/t/accessible-from-remote-computer/236
 - https://forum.activitywatch.net/t/activitywatch-as-employee-monitoring-software/499
